src/System.Private.CoreLib/src/System/Reflection/FieldAccessor.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using Internal.Runtime.CompilerServices;
+
+#if BIT64
+using nuint = System.UInt64;
+#else
+using nuint = System.UInt32;
+#endif
+
+namespace System.Reflection
+{
+    public sealed class FieldAccessor<TObject, TField> where TObject : class
```

Comment:
Why is `FieldAccessor` not a struct?

Reply:
Would need to add an initalized field to ensure it had sensible values for `default(FieldAccessor)`
```csharp
private readonly uint _fieldOffset;
private readonly bool _initalized;
```
and additionally need  a check in the GetRef
```csharp
public ref TField GetRef(TObject obj)
{
    if (!_initalized) ThrowHelper.ThrowNotInitalized();
    // ...
}
```
Question is does it add value in terms of the extra cost vs size saved in number of instances and one less indirection.

Reply:
Yeah, Ben and I had a brief conversation on this the other day. The type wouldn't be safe to use if it were a struct, as not only would uninitialized usage be a problem, the struct would also be subject to tearing concerns.

There's no real perf hit to making this a class instead of a struct. The intent is that the caller will cache the instance and reuse it. Since the accessor itself is inlined and usage doesn't incur a null check on the accessor itself (see the comment at the top of `GetRef`), this should take care of any perf concerns.